src/m365/spo/commands/listitem/listitem-list.ts

```diff
@@ -227,13 +265,53 @@ class SpoListItemListCommand extends SpoCommand {
         data: requestBody
       };
 
-      const listItemInstances = args.options.camlQuery ? await request.post<ListItemInstanceCollection>(requestOptions) : await request.get<ListItemInstanceCollection>(requestOptions);
-      listItemInstances.value.forEach(v => delete v['ID']);
-      logger.log(listItemInstances.value);
+      const listItemInstances = await request.post<ListItemInstanceCollection>(requestOptions);
+      skipTokenId = listItemInstances.value.length > 0 ? listItemInstances.value[listItemInstances.value.length - 1].Id : undefined;
+      items.push(...listItemInstances.value);
     }
-    catch (err: any) {
-      this.handleRejectedODataJsonPromise(err);
+    while (skipTokenId !== undefined);
+
+    return items;
+  }
+
+  private getExpandFieldsArray(fieldsArray: string[]): string[] {
+    const fieldsWithSlash: string[] = fieldsArray.filter(item => item.includes('/'));
+    const fieldsToExpand: string[] = fieldsWithSlash.map(e => e.split('/')[0]);
+    const expandFieldsArray: string[] = fieldsToExpand.filter((item, pos) => fieldsToExpand.indexOf(item) === pos);
+    return expandFieldsArray;
+  }
+
+  private async getFormDigestValue(logger: Logger, webUrl: string): Promise<string> {
+    if (this.verbose) {
+      logger.logToStderr(`Getting request digest for query request`);
     }
+
+    const response = await spo.getRequestDigest(webUrl);
+    return response.FormDigestValue;
+  }
+
+  private async getSkipTokenId(logger: Logger, options: Options, listApiUrl: string): Promise<number | undefined> {
```

Comment:
Let's clarify this name. We're not retrieving the skip token here, but rather the ID of the last item that belongs to the specified page.